fastlane_core/lib/fastlane_core/project.rb

```diff
@@ -438,22 +438,27 @@ def self.run_command(command, timeout: 0, retries: 0, print: true)
 
       total_tries = retries + 1
       try = 1
+      try_timeout = timeout
       begin
-        Timeout.timeout(timeout) do
+        Timeout.timeout(try_timeout) do
           # Using Helper.backticks didn't work here. `Timeout` doesn't time out, and the command hangs forever
           result = `#{command}`.to_s
         end
       rescue Timeout::Error
         try_limit_reached = try >= total_tries
 
-        message = "Command timed out after #{timeout} seconds on try #{try} of #{total_tries}"
-        message += ", trying again..." unless try_limit_reached
+        # Try harder on each iteration
+        next_timeout = try_timeout * 2
+
+        message = "Command timed out after #{try_timeout} seconds on try #{try} of #{total_tries}"
+        message += ", trying again with #{next_timeout} second timeout..." unless try_limit_reached
```

Comment:
typo "seconds" ?

Reply:
Thanks for taking a look. I updated the method's documentation with details about the timeout doubling behavior. I don't know if you thought "second" should be "seconds", but I think it would sound awkward to say e.g. "trying again with 30 seconds timeout". I tweaked it to put "a" before the number for e.g. "trying again with a 30 second timeout". Maybe that is more clear?